sdk/attestation/azure-security-attestation/src/attestation_administration_client.cpp

```diff
@@ -126,6 +122,163 @@ AttestationAdministrationClient::GetAttestationPolicy(
   return Response<AttestationToken<std::string>>(returnedToken, std::move(response));
 }
 
+Models::AttestationToken<std::nullptr_t>
+AttestationAdministrationClient::CreateSetAttestationPolicyToken(
+    Azure::Nullable<std::string> const& newAttestationPolicy,
+    Azure::Nullable<AttestationSigningKey> const& signingKey) const
+{
+  // Embed the encoded policy in the StoredAttestationPolicy.
+  Azure::Nullable<StoredAttestationPolicy> storedPolicy;
+  if (newAttestationPolicy)
+  {
+    storedPolicy = StoredAttestationPolicy{std::vector<uint8_t>(
+        newAttestationPolicy.Value().begin(), newAttestationPolicy.Value().end())};
+  }
+
+  auto tokenToSet(
+      AttestationTokenInternal<StoredAttestationPolicy, StoredAttestationPolicySerializer>::
+          CreateToken(storedPolicy, signingKey));
+  auto tokenToSend(static_cast<AttestationToken<StoredAttestationPolicy>>(tokenToSet));
+
+  // Strip the body type off the returned JWS - the caller of the function doesn't need it.
+  return AttestationTokenInternal<std::nullptr_t>(tokenToSend.RawToken);
```

Comment:
Template specialization needs a type, unless you're suggesting that I use `decltype{nullptr)`? What I'm trying to do is to create an AttestationToken<> - in other words, something with a `Body` property that cannot be used.

Reply:
No, 
I mean that, for an empty response, instead of having something like `Response<void>` or `Response<std::nullptr>` to represent a `Response<T>`,  we would create an empty struct `Name` and use it to return `Response<Name>` where `Name` is just a `struct Name {};` ( no fields or anything.

Reply:
The reason I'm using nullptr_t is that the Body field in the AttestationToken specialization cannot be accessed without generating a compiler error. If I use an empty structure, then the field can still be accessed.

In my optimal solution, I would have a class and a specialization:

```c++
class AttestationToken 
{ 
<attestation token specific fields, but no "Body" field>
}

template<typename T>
class AttestationToken<T> : public AttestationToken
{
  T Body;
}
```

Then you would either have an AttestationToken or an AttestationToken<T> and deal with them appropriately. Unfortunately C++ doesn't allow this construct, so I have to improvise something that gives close to the same functionality.

Using `nullptr_t` for the specialization type gives me a very close approximation to what I want the developer experience to be.